contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,603 @@
+        }
+        swapData.amount = swapData.totalHackersHatReward + governanceHatReward[_asset];
+        if (swapData.amount == 0) revert AmountToSwapIsZero();
+        IERC20 _HAT = HAT;
+        (swapData.hatsReceived, swapData.amountUnused) = _swapTokenForHAT(IERC20(_asset), swapData.amount, _amountOutMinimum, _routingContract, _routingPayload);
+
+        governanceHatReward[_asset] = swapData.amountUnused * governanceHatReward[_asset] / swapData.amount;
+
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            uint256 hackerReward = swapData.hatsReceived * hackersHatReward[_asset][_beneficiaries[i]] / swapData.amount;
+            swapData.totalHackerReward += hackerReward;
+            hackersHatReward[_asset][_beneficiaries[i]] = swapData.amountUnused * hackersHatReward[_asset][_beneficiaries[i]] / swapData.amount;
+            address tokenLock;
+            if (hackerReward > 0) {
+                // hacker gets her reward via vesting contract
+                tokenLock = tokenLockFactory.createTokenLock(
+                    address(_HAT),
+                    0x0000000000000000000000000000000000000000, //this address as owner, so it can do nothing.
+                    _beneficiaries[i],
+                    hackerReward,
+                    // solhint-disable-next-line not-rely-on-time
+                    block.timestamp, //start
+                    // solhint-disable-next-line not-rely-on-time
+                    block.timestamp + generalParameters.hatVestingDuration, //end
+                    generalParameters.hatVestingPeriods,
+                    0, // no release start
+                    0, // no cliff
+                    ITokenLock.Revocability.Disabled,
+                    true
+                );
+                _HAT.safeTransfer(tokenLock, hackerReward);
+            }
+            emit SwapAndSend(_beneficiaries[i], swapData.amount, hackerReward, tokenLock);
+        }
+        _HAT.safeTransfer(owner(), swapData.hatsReceived - swapData.totalHackerReward);
+    }
+
+    /**
+    * @notice Returns the general parameters for all vaults
+    * @return generalParameters: See { HATVaultsRegistry-generalParameters }
+    */    
+    function getGeneralParameters() external view returns(GeneralParameters memory) {
+        return generalParameters;
+    }
+
+    /**
+    * @notice Returns the number of vaults that have been previously created 
+    */
+    function getNumberOfVaults() external view returns(uint256) {
+        return hatVaults.length;
+    }
+
+    /** 
+    * @dev Check that a given hats bounty split is legal, meaning that:
+    *   Each entry is a number between 0 and less than `HUNDRED_PERCENT`.
+    *   Total splits should be less than `HUNDRED_PERCENT`.
+    * function will revert in case the bounty split is not legal.
+    * @param _hatBountySplit The bounty split to check
+    */
+    function validateHATSplit(HATBountySplit memory _hatBountySplit) public pure {
```

Comment:
Given that this is governance setting it's own fee, perhaps this should be limited to a maxium?